drivers/sensor/kx022/kx022_trigger.c

```diff
@@ -0,0 +1,315 @@
+	data->hw_tf->write_reg(data, KX022_REG_ATH, KX022_ATH_THS);
+
+	if (KX022_DEFAULT_INT_PIN == KX022_INT1) {
+		data->hw_tf->update_reg(data, KX022_REG_INC4, KX022_MAKS_INC4_WUFI1,
+					KX022_INC4_WUFI1);
+	} else if (KX022_DEFAULT_INT_PIN == KX022_INT2) {
+		data->hw_tf->update_reg(data, KX022_REG_INC6, KX022_MAKS_INC6_WUFI2,
+					KX022_MAKS_INC6_WUFI2);
+	}
+	kx022_mode(dev, KX022_OPERATING_MODE);
+}
+
+void kx022_drdy_setup(const struct device *dev, sensor_trigger_handler_t handler)
+{
+	struct kx022_data *data = dev->data;
+
+	data->drdy_handler = handler;
+	kx022_mode(dev, KX022_STANDY_MODE);
+	data->hw_tf->update_reg(data, KX022_REG_CNTL1, KX022_MASK_CNTL1_DRDYE, KX022_CNTL1_DRDYE);
+	data->hw_tf->update_reg(data, KX022_REG_ODCNTL, KX022_MASK_ODCNTL_OSA, KX022_ODCNTL_50HZ);
+
+	if (KX022_DEFAULT_INT_PIN == KX022_INT1) {
+		data->hw_tf->update_reg(data, KX022_REG_INC4, KX022_MAKS_INC4_DRDYI1,
+					KX022_INC4_DRDYI1);
+	}
+	if (KX022_DEFAULT_INT_PIN == KX022_INT2) {
+		data->hw_tf->update_reg(data, KX022_REG_INC6, KX022_MAKS_INC6_DRDYI2,
+					KX022_INC6_DRDYI2);
+	}
+
+	kx022_mode(dev, KX022_OPERATING_MODE);
+}
+
+void kx022_Tilt_setup(const struct device *dev, sensor_trigger_handler_t handler)
+{
+	struct kx022_data *data = dev->data;
+
+	data->slope_handler = handler;
+
+	kx022_mode(dev, KX022_STANDY_MODE);
+
+	data->hw_tf->update_reg(data, KX022_REG_CNTL1, KX022_MASK_CNTL1_TPE, KX022_CNTL1_TPE_EN);
+
+	k_msleep(50);
+	data->hw_tf->write_reg(data, KX022_REG_CNTL2, KX022_CNTL_TILT_ALL_EN);
+	k_msleep(50);
+
+	data->hw_tf->update_reg(data, KX022_REG_CNTL3, KX022_MASK_CNTL3_OTP,
+				(KX022_DEFAULT_TILT_ODR << KX022_OTP_SHIFT));
+	k_msleep(50);
+
+	data->hw_tf->write_reg(data, KX022_REG_TILT_TIMER, KX022_TILT_TIMES_DURATION);
+	k_msleep(50);
+
+	data->hw_tf->write_reg(data, KX022_REG_TILT_ANGLE_LL, KX022_DEF_TILT_ANGLE_LL);
+	k_msleep(50);
+	data->hw_tf->write_reg(data, KX022_REG_TILT_ANGLE_HL, KX022_DEF_TILT_ANGLE_HL);
+	k_msleep(50);
+	data->hw_tf->write_reg(data, KX022_REG_HYST_SET, KX022_DEF_HYST_SET);
+	k_msleep(50);
+
+	if (KX022_DEFAULT_INT_PIN == KX022_INT1) {
+		data->hw_tf->update_reg(data, KX022_REG_INC4, KX022_MAKS_INC6_TPI2,
+					KX022_INC4_TPI1);
+	} else if (KX022_DEFAULT_INT_PIN == KX022_INT2) {
+		data->hw_tf->update_reg(data, KX022_REG_INC6, KX022_MAKS_INC6_TPI2,
+					KX022_INC6_TPI2);
+	}
+
+	kx022_mode(dev, KX022_OPERATING_MODE);
+	k_msleep(100);
+}
+
+void kx022_clear_setup(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+
+	kx022_mode(dev, KX022_STANDY_MODE);
+	data->hw_tf->write_reg(data, KX022_REG_CNTL1, KX022_DEFAULT_CNTL1);
+	kx022_mode(dev, KX022_OPERATING_MODE);
+	k_msleep(50);
+}
+int kx022_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+		      sensor_trigger_handler_t handler)
+{
+	struct kx022_data *data = dev->data;
+	const struct kx022_config *cfg = dev->config;
+	uint8_t buf[6];
+
+	switch (trig->type) {
```

Comment:
Can do `kx022_mode(dev, KX022_STANDY_MODE);` before switch and `kx022_mode(dev, KX022_OPERATING_MODE);` before return or break.